packages/nextjs/test/userIntegrations.test.ts

```diff
@@ -0,0 +1,53 @@
+import { RewriteFrames } from '@sentry/integrations';
+import { Integration } from '@sentry/types';
+
+import { addIntegration } from '../src/utils/userIntegrations';
+
+const testIntegration = new RewriteFrames();
+type UserFunctionIntegrations = (integrations: Integration[]) => Integration[];
+
+describe('user integrations without any integrations', () => {
+  test('as an array', () => {
+    const userIntegrations: Integration[] = [];
+    // Should get a single integration
+    let finalIntegrations = addIntegration(testIntegration, userIntegrations);
+    expect(finalIntegrations).toBeInstanceOf(Array);
+    finalIntegrations = finalIntegrations as Integration[];
+    expect(finalIntegrations).toHaveLength(1);
+    expect(finalIntegrations[0]).toMatchObject(testIntegration);
+  });
+
+  test('as a function', () => {
+    const userIntegrationFnc: UserFunctionIntegrations = (): Integration[] => {
+      return [];
+    };
+    // Should get a single integration
+    const integrationWrapper = addIntegration(testIntegration, userIntegrationFnc);
+    expect(integrationWrapper).toBeInstanceOf(Function);
+    const finalIntegrations = (integrationWrapper as UserFunctionIntegrations)([]);
+    expect(finalIntegrations).toHaveLength(1);
+    expect(finalIntegrations[0]).toMatchObject(testIntegration);
+  });
+});
+
+describe('user integrations with integrations', () => {
+  test('as an array', () => {
+    const userIntegrations = [new RewriteFrames()];
+    // Should get the same array (with no patches)
+    const finalIntegrations = addIntegration(testIntegration, userIntegrations);
+    expect(finalIntegrations).toMatchObject(userIntegrations);
+  });
+
+  test('as a function', () => {
+    const userIntegrations = [new RewriteFrames()];
+    const integrationsFnc: UserFunctionIntegrations = (_integrations: Integration[]): Integration[] => {
+      return userIntegrations;
+    };
+    // Should get a function that returns the test integration
+    let finalIntegrations = addIntegration(testIntegration, integrationsFnc);
+    expect(typeof finalIntegrations === 'function').toBeTruthy();
```

Comment:
`toBeTruthy` does type coercion by doing `!!expectValue`, which I don't think we need here, prefer `toBe(true)` for increased clarity.

```suggestion
    expect(typeof finalIntegrations === 'function').toBe(true);
```

Reply:
I'll repeat my feedback from [the original PR](https://github.com/getsentry/sentry-javascript/pull/3375#discussion_r610157980) that I think we ought to save `.toBe(true)` for times when we're actually testing a boolean value. (Otherwise, why not make every test `expect(<some assertion>).toBe(true)`?) I'd much rather see this as `expect(typeof finalIntegrations).toEqual("function")`, both for that reason and because then it's immediately obvious that function-ness is what we're testing here.

I get that this has already been merged, but in future test-writing, I'd love it if this is something we could keep in mind.